### DISCUSSION TITLE
Referrer and UTM tags

### DISCUSSION THREAD

**Discussion by Pyroarsonist** [microsoft/vscode-discussions#1355] [Extension Development QnA] [answered]
Does VSCode Marketplace supports referer links or UTM tags? Maybe some custom query parameters could be used?
For example, link _https://marketplace.visualstudio.com/items?itemName=extension_name&referrer=some_referrer_ would have have `referrer` = _some_referrer_.
This is needed to understand from which place vscode extension was being installed.

**Comment by robconery** [comment, marked as answer]
If I understand correctly, you want to know where your extension is being installed from (domain, IP, etc)? Inspecting the query string in this way is only something that `marketplace.visualstudio.com` can do given that it's the domain processing the URL, not your extension. 

If you need telemetry, you can read more here: https://code.visualstudio.com/api/extension-guides/telemetry.
